### DISCUSSION TITLE
PI4B with touch display - White screen instead of LBB UI.

### DISCUSSION THREAD

**Discussion by jclarke0000** [outdoorbits/little-backup-box#329] [Q&A] [answered]
This is a new problem I discovered yesterday.  I made a new install on a Pi 4B/8GB ram with a 4/3" Waveshare capacitive touch screen attached. The problem arises when the LBB software attempts to start.  The full-screen window appears, but it is blank white with dark grey rectangle across the top menu bar area.  This screen will stay on indefinitely unless I connect a keyboard and press Alt-Tab.  When I do that, the screen changes to fully white -- no grey menu bar rectangle -- and then after a few seconds the LBB UI appears.

This is not the Comitup UI as I have the same problem if the Pi can't connect to a known WiFi.  If I press Alt-Tab, I get the same transition to a fully white screen followed by the Comitup UI appearing as normal.

This is a new problem that has crept in between yesterday and a couple of weeks ago.  I had a previous install with a different touch screen, and if I flash my card with that backup, I get the UI showing up on the new screen after a normal boot.  So it's either something has changed with Pi OS or with LBB's software that is causing this.

To be clear, the software IS running and working -- if I connect from a different device -- say my phone's browser -- I can see the UI and run a backup like normal.  It simply appears that the UI is not opening in the foreground on the device itself.

I've attached a screen shot of the blank screen as well as a video showing what I see on the screen from cold boot

![IMG_0319](https://github.com/user-attachments/assets/fa989fa3-79ce-4a11-afd9-8632ef1f8e65)

https://github.com/user-attachments/assets/6e998360-07ab-48d8-8ffc-0aa304b4b5a0

Here are the Pi OS details
```
pi@lbb:~ $ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

pi@lbb:~ $ cat /etc/debian_version
12.7
```

When I go to the System tab I DO see a status error:
```
ERROR: smartctl requires a device name as the final command-line argument.
```

Related?

Otherwise, any Ideas?

Thanks!
 - Jeff

**Comment by jclarke0000** [comment]
More on this.  I tracked this down to the autostart script found here:

```
/home/lbb-desktop/.config/autostart/little-backup-box.desktop
```
On the line that launches firefox:

```
Exec=firefox -setDefaultBrowser --kiosk http://localhost
```

It's the `--kiosk` flag that's causing problems.  If I remove it, Firefox starts correctly with the UI.  Of course it's not in full screen, but it does work.

**Comment by outdoorbits** [comment]
But full screen is necessary, isn't it? Maybe you figure out a fullscreen solution? I'll be back next week and also try to fix that

**Comment by jclarke0000** [comment]
I tinkered around with this for a while this evening, and this worked for me:
```
Exec=chromium-browser --kiosk http://localhost
```

I needed to use Chromium instead of Firefox AND I needed to remove the `-setDefaultBrowser` flag.  No idea why this secret sauce works, but it does.  This solves my issue.

**Comment by outdoorbits** [comment]
Great! I'll change that next week.

**Reply by jclarke0000** [reply]
More on this.  I've discovered there's an unintended side-effect of the above -- When I insert the USB device to initiate a backup I get the OS popup asking what action I'd like to take.  This is a simple matter of choose the "do nothing" option and checking the "Always do this" box in order to prevent the popup from appearing again, but it's exposing a flaw in that using chromium with the `--kiosk` flag doesn't keep these popups from appearing.  I'm wondering if there's another startup flag that needs to be used in order to do so...

**Comment by outdoorbits** [comment]
I just uploaded a new version having comitup integrated into the web UI. The desktop now loads the web UI any time. To access comitup you have to follow the menu link (only visible if the hotspot is active). Because of the side effects of chromium-browser and --kiosk I did change this yet. Would you please test it?

**Reply by jclarke0000** [reply]
~~How do I perform an update without setting a global password? I don't like how the global password prevents the UI from loading without providing the password, so I haven't set one.~~

After all of the steps below, I realized there was an option to remove the password after it has been set.

**Reply by jclarke0000** [reply]
Also, Since I have a backup I went ahead and set a global password just to try, but now I am locked out. the user name `lbb` with the password I set isn't accepted, nor is the user name `lbb-desktop`.  I even tried the user name `pi` and that didn't work either.

**Reply by jclarke0000** [reply]
In fact, I can't even SSH into the box anymore.  I'm trying to sign in with `pi` and the password I set (Which is the same as the password for user `pi` previously).

**Reply by jclarke0000** [reply]
More on this.  I've discovered that LBB doesn't like passwords with special characters in them.  I tried setting a password with a `$` symbol and an `@` symbol, and this caused me to be locked out of LBB, including through SSH.  If I instead set a password with `S` instead of `$` and `A` instead of `@` I have no problems logging in afterwards.

**Comment by jclarke0000** [comment]
After performing the update, I cleared all of Comitup's credentials, and after a reboot, I'm still getting the Comitup UI displaying at startup on the device.  See screenshot:
![IMG_0372](https://github.com/user-attachments/assets/d11828fd-efba-49f6-a186-a8a9d3017525)

Let me know if you need me to provide any log files.

**Reply by jclarke0000** [reply, marked as answer]
One more on this:  I was able to solve it be changing my autostart script from this:

`Exec=chromium-browser --kiosk http://localhost`

To this:

`Exec=chromium-browser --kiosk http://localhost:8080`

**Comment by outdoorbits** [comment]
Just repeat the command from installation. The script will run in update 
mode.

**Comment by outdoorbits** [comment]
Due limitations of bash config files, some characters are not accepted. 
In an upcoming commit I'll allow most special characters but not quotes, 
spaces and backslashes.

**Comment by outdoorbits** [comment]
About the white screen surprisingly the was a difference between HDMI connected monitors and those connected to the screen port. This is fixed now so please update and tell me. 
The other problem about the passwords should be solved also.

**Reply by jclarke0000** [reply]
@outdoorbits  Sorry for the belated response on this.  I installed the update and the issue is fixed 

BUT!  The update did NOT recreate the `/home/lbb-desktop/.config/autostart/little-backup-box.desktop` file.  I needed to manually update it to match the setup script.

From `setup-graphical-environment.sh` I copied the line `Exec=firefox -setDefaultBrowser -private --kiosk=http://localhost:8080` and pasted in into my existing autostart script, and after a reboot, the the problem was gone.

I would imagine anyone installing this from scratch won't have this issue.

**Reply by jclarke0000** [reply]
Also, it appears that the update also modified the hostname.  I previously had the hostname set as `lbb`, but after the update it was set to `little-black-box`

**Comment by outdoorbits** [comment]
In my tests it did. Did you try this in a native lbb environment? What's the owner and the permissions of that file (/home/lbb-desktop/.config/autostart/little-backup-box.desktop)?

**Reply by jclarke0000** [reply]
owner of the file is root:root, permissions are `-rw-r--r--`.  This was the one I modified, but I kept a backup of the original, whose owner is lbb-desktop:lbb-desktop.  Likely this was the problem.

**Comment by outdoorbits** [comment]
Yes, I changed this from the native hostname. Any issues about that?

**Reply by jclarke0000** [reply]
No issues really... threw me for a loop though when I couldn't connect over SSH after the update using the hostname.  After connecting with the IP address instead, that's when I realized the hostname had been changed.  I liked using the hostname `lbb` as it is easier to type than `little-backup-box` when I connect over SSH.  I've changed it back to `lbb`.  Will there be any issues with that?

**Reply by outdoorbits** [reply]
Ok, I understand. So let's change the default hostname to lbb.
